src/read.c

```diff
@@ -3823,11 +3823,24 @@ static avifResult avifParse(avifDecoder * decoder)
         if (!memcmp(header.type, "ftyp", 4) || !memcmp(header.type, "meta", 4) || !memcmp(header.type, "moov", 4)) {
 #endif
             boxOffset = parseOffset;
-            readResult = decoder->io->read(decoder->io, 0, parseOffset, header.size, &boxContents);
+            size_t sizeToRead;
+            if (header.size == 0) {
+                // A size of 0 means the box body goes till the end of the file.
+                if (decoder->io->sizeHint != 0 && decoder->io->sizeHint - parseOffset < SIZE_MAX) {
```

Comment:
Prefer `decoder->io->sizeHint  < parseOffset + SIZE_MAX` as you are dealing with uint64_t (to prevent casts to int64_t)

Reply:
How can `uint64_t - uint64_t < size_t` lead to int64_t?  
Also `parseOffset + SIZE_MAX` may overflow.

Reply:
Ah, all is cast to uint64_t so that's fine indeed. And I missed the `AVIF_ASSERT_OR_RETURN(decoder->io->sizeHint == 0 || parseOffset <= decoder->io->sizeHint);` the above.